src/preprocessing/preprocessing_pass_context.h

```diff
@@ -81,11 +81,21 @@ class PreprocessingPassContext
   void recordSymbolsInAssertions(const std::vector<Node>& assertions);
 
   /**
-   * Add substitution to theory model.
+   * Add substitution to theory model. This method should only be called if
+   * we have already added the substitution to the top-level substitutions
+   * class. Otherwise, addSubstitution should be called instead.
```

Comment:
If there is a way to partially check this using an Assert in the code, that would be nice.

Reply:
Not at the moment, due to the order in which things occur.  In the end, we need to clean up calls to this method so that everything calls methods for adding substitutions that ensure this invariant is maintained.